src/main/java/com/adyen/ApiKeyAuthenticatedService.java

```diff
@@ -0,0 +1,33 @@
+/*
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Java API Library
+ *
+ * Copyright (c) 2018 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ */
+package com.adyen;
+
+/**
+ * <h1>ApiKeyAuthenticatedService</h1>
+ * The ApiKeyAuthenticatedService is an interface to enable a child service to support API key authentication feature.
+ */
+public class ApiKeyAuthenticatedService extends Service {
+
+    protected ApiKeyAuthenticatedService(Client client) {
+        super(client);
+        setApiKeySupported(true);
```

Comment:
let's better rename to setApiKeyRequired

Reply:
done!